File: TsinghuaNet.Eto/Converters/StringFormatConverter.vb
Imports System.Globalization
Imports Eto.Forms

Public Class StringFormatConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If parameter Is Nothing Then
            Return value.ToString()
        Else
            Dim format = parameter.ToString()
            If String.IsNullOrEmpty(format) Then
                Return value.ToString()
            Else
                Return String.Format(format, value)
            End If
        End If
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

File: TsinghuaNet.Eto/TsinghuaNet.Eto/Converters/StringFormatConverter.cs
﻿using System;
using System.Globalization;
using System.Threading;
using Eto.Forms;

namespace TsinghuaNet.Eto.Converters
{
    public class StringFormatConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var format = parameter?.ToString();
            if (culture == CultureInfo.InvariantCulture)
                culture = Thread.CurrentThread.CurrentCulture;
            if (string.IsNullOrEmpty(format))
                return string.Format(culture, "{0}", value);
            else
                return string.Format(culture, format, value);
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}
